torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py

```diff
@@ -2,6 +2,18 @@
 import torch.distributed as dist
 
 
+def allreduce_fut(
+    process_group: dist.ProcessGroup, tensor: torch.Tensor
+) -> torch.futures.Future:
+    "Averages the input gradient tensor by allreduce and returns a future."
+    fut = dist.all_reduce(tensor, group=process_group, async_op=True).get_future()
+
+    def div_by_group_size(fut):
+        return [fut.value()[0].div_(process_group.size())]
```

Comment:
Is it  possible for `process_group` to be `None`? Should we add an assert if not?

Reply:
Good idea! Duplicated `group_to_use = process_group if process_group is not None else dist.group.WORLD` inside this util function.